code/modules/cm_aliens/weeds.dm

```diff
@@ -199,12 +199,20 @@
 	for(var/obj/structure/platform/P in src.loc)
 		if(P.dir == reverse_direction(direction))
 			return FALSE
+	for(var/obj/structure/barricade/B in loc)
+		if(B.density && B.dir == direction && B.health >= (B.maxhealth / 4))
+			return FALSE
```

Comment:
rename B to a more descriptive variable name, 'barricade' if nothing else.

Reply:
done